.github/workflows/lint-ui.yaml

```diff
@@ -24,8 +24,11 @@ jobs:
       - name: Install Yarn dependencies
         run: yarn --frozen-lockfile
 
+      - name: Run Type Check
+        run: yarn type-check
```

Comment:
I reordered the jobs so that the type check runs before Prettier and ESLint. Type checking is arguably more important – the information that the code is formatted correctly and adheres to project conventions is useless if it fundamentally doesn't type check.

OTOH I'm not in cases where teleport.e is broken maybe it'd actually be helpful to first see the lint errors and keep the type check at the end. But teleport.e has its own lint-ui.yaml and its own version of this PR (https://github.com/gravitational/teleport.e/pull/2756).

Reply:
Running the type check first sounds good.